src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsUapHiViMobileWiki.java

```diff
@@ -0,0 +1,274 @@
+    AssertionAds.assertAspectRatio(slot.getSize(), IMPACT_STATE_ASPECT_RATIO);
+  }
+
+  @Test
+  public void shouldHaveCorrectAspectRatioForResolvedState() {
+    openPageOnSecondPV();
+    final WebElement slot = (new HiViUap(driver, TLB_SLOT_ID)).waitForAdLoaded();
+    AssertionAds.assertAspectRatio(slot.getSize(), RESOLVED_STATE_ASPECT_RATIO);
+  }
+
+  @Test
+  public void shouldHaveResolvedStateAfterScroll() throws InterruptedException {
+    AdsBaseObject adsPage = openPage();
+    WebElement slot = driver.findElement(TLB_SELECTOR);
+    int defaultStateHeight = slot.getSize().getHeight();
+    int scrollBy = 50;
+
+    AssertionAds.assertAspectRatio(slot.getSize(), IMPACT_STATE_ASPECT_RATIO);
+
+    adsPage.scrollBy(0, scrollBy);
+    TimeUnit.SECONDS.sleep(1);
+    Assertion.assertEquals(slot.getSize().getHeight(), defaultStateHeight - scrollBy);
+
+    adsPage.scrollBy(0, 500);
+    TimeUnit.SECONDS.sleep(1);
+    AssertionAds.assertAspectRatio(slot.getSize(), RESOLVED_STATE_ASPECT_RATIO);
+  }
+
+  @Test
+  public void shouldKeepResolvedStateAspectRatioAfterScroll() throws InterruptedException {
+    AdsBaseObject adsPage = openPageOnSecondPV();
+
+    HiViUap hivi = new HiViUap(driver, TLB_SLOT_ID);
+    WebElement slot = hivi.waitForAdLoaded();
+
+    AssertionAds.assertAspectRatio(slot.getSize(), RESOLVED_STATE_ASPECT_RATIO);
+
+    adsPage.scrollBy(0, 500);
+    TimeUnit.SECONDS.sleep(1);
+    AssertionAds.assertAspectRatio(slot.getSize(), RESOLVED_STATE_ASPECT_RATIO);
+
+    adsPage.scrollBy(0, -500);
+    TimeUnit.SECONDS.sleep(1);
+    AssertionAds.assertAspectRatio(slot.getSize(), RESOLVED_STATE_ASPECT_RATIO);
+  }
+
+  @Test
+  public void shouldAutoplayVideoForImpactState() {
+    openPage();
+    HiViUap hiViUap = new HiViUap(driver, TLB_SLOT_ID);
+    hiViUap.waitForVideoStart();
+  }
+
+  @Test
+  public void shouldProgressTimeWhilePlaying() throws InterruptedException {
+    openPage();
+
+    HiViUap hiViUap = new HiViUap(driver, TLB_SLOT_ID);
+    hiViUap.waitForVideoStart();
+
+    hiViUap.clickVideo();
+    final double startProgressBarWidth = hiViUap.getProgressBarWidth();
+    TimeUnit.SECONDS.sleep(3);
+    hiViUap.clickVideo();
+
+    Assert.assertTrue(startProgressBarWidth < hiViUap.getProgressBarWidth(), "Video time indicator should move.");
+  }
+
+  @Test
+  public void shouldMuteVideoForAutoplayedImpactState() throws InterruptedException {
+    openPage();
+    new HiViUap(driver, TLB_SLOT_ID);
+
+    TimeUnit.SECONDS.sleep(3);
+    Assertion.assertFalse(SoundMonitor.wasSoundHeardOnPage(new JavascriptActions()));
+    TimeUnit.SECONDS.sleep(3);
+    Assertion.assertFalse(SoundMonitor.wasSoundHeardOnPage(new JavascriptActions()));
+  }
+
+  @Test
+  public void shouldUnmuteVideoAfterClickOnIcon() throws InterruptedException {
+    openPage();
+    HiViUap hiViUap = new HiViUap(driver, TLB_SLOT_ID);
+
+    hiViUap.waitForVideoStart();
+    hiViUap.clickVideo();
+    hiViUap.toggleSound();
```

Comment:
It's interesting it works with Porvata but doesn't work with JWPlayer (toggle sound with NEXUS_5 emulation).

Reply:
We probably know [more about it](https://github.com/Wikia/selenium-tests/pull/2156#discussion_r164689800).